py/terra_widgets/workspace_paths.py

```diff
@@ -0,0 +1,104 @@
+"""Methods to obtains paths to files within the workspace bucket."""
+
+import datetime
+import fnmatch
+import os
+from typing import Dict
+from typing import List
+from typing import NamedTuple
+
+
+WorkspaceDestination = NamedTuple('WorkspaceDestination', [('html_file', str), ('comment_file', str)])
+
+
+class WorkspacePaths:
+  """Encapsulate all logic for manipulating workspace paths.
+
+  Paths are of the form:
+    gs://fc-secure-<guid>/reports/<user>@researchallofus.org/<date>/<time>/<notebook>.html
+    gs://fc-secure-<guid>/reports/<user>@researchallofus.org/<date>/<time>/<notebook>.html.comment.txt
```

Comment:
I don't think I'd change this, as it isn't a problem for AoU but food for thought: for Terra in general, it is likely prudent to avoid putting user emails directly in paths or autogenerated files/metadata which could be visible to readers. I did some work on this as part of the locking design: https://docs.google.com/document/d/1ay45skd9wxMqtoHZdfUlVAB0b-OJYBZhTGQKFCIcYuM/edit

Of course, as a client you don't have much of an option, so ideally Terra would provide some kind of standard mechanism here.

Reply:
They way I addressed it was to say front and center within the user interface that it uses your email address as part of the path.